src/garage/tf/models/gaussian_cnn_model.py

```diff
@@ -270,14 +271,14 @@ def _build(self, state_input, name=None):
                         layer_normalization=self._layer_normalization)
                 else:
                     batch_dim = tf.shape(state_input)[0]
-                    log_std_network, _ = parameter(
-                        state_input,
+                    log_std_param = parameter(
                         length=action_dim,
-                        batch_dim=[batch_dim],
                         initializer=tf.constant_initializer(
                             self._init_std_param),
                         trainable=self._learn_std,
                         name='log_std_network')
+                    log_std_network = broadcast(
+                        log_std_param, state_input, batch_dim=[batch_dim])
```

Comment:
why would batch_dim be a list?

Reply:
The helper function expects batch_dim to be a list since it can be a list (for recurrent policies).

Reply:
We can make it accept an integer, and turn into a list tho

Reply:
How about this?

```
def broadcast_with_batch(param, input_var, is_time_series=False):
    if is_time_series:
        batch_dim = tf.shape(input_var)[:-1]
    else:
        batch_dim = tf.shape(input_var)[0]
    broadcast_shape = tf.concat(axis=0, values=[batch_dim, tf.shape(param)])
    return tf.broadcast_to(param, shape=broadcast_shape)
```

I think we only have this two cases.

Reply:
@ryanjulian Is it better?

Reply:
If this looks ok, I will also rename it to `broadcast_with_batch`

Reply:
i suggest you add a param `batch_dim` instead. switching function behavior based on a type-specific bool is an anti-pattern.

either we should have two different functions for two different types (time series batches and non time-series batches), or we should parameterize the possible cases and ignore the types (e.g. `batch_dim`).

it would be nice if we could just unify the batch position for all data. is there really a need for having time series reverse their batch position? in any case, fixing that globally is a better candidate for an issue than this PR.

Reply:
err i see. you actually implemented my suggestion above, but provided a different one as example.

Reply:
Should we define `batch_dim` as the first `batch_dim` dimension as the batch dimension? Then it becomes
```python
def broadcast_with_batch(param, input_var, batch_dim=1):
        assert batch_dim > 0
        batch_shape = tf.shape(input_var)[:batch_dim]
    broadcast_shape = tf.concat(axis=0, values=[batch_shape, tf.shape(param)])
    return tf.broadcast_to(param, shape=broadcast_shape)
``` 
For non-recurrent primitives, we have `batch_dim=1` while for recurrent policies, we will have `batch_dim=2`.

Reply:
i think batch dims should be a tuple or list of dimensions which are considered part of the batch. for recurrent, you told me there is more than one batch dimension, so you would list those in batch_dims

e.g.
```python
# if dimensions 3 and 4 are considered batch
foo = broadcast_with_batch(bar, bux, batch_dims=(3,4))
```

you might also consider for convenience allowing non-iterable inputs for convenience
```
# dimension 0 is the batch
foo = broadcast_with_batch(bar, bux, batch_dims=0)
```

by the way, your example above uses default `batch_dim=1` -- aren't dimensions zero-indexed? what data uses 1 as the batch dimension?